### DISCUSSION TITLE
Images attachment

### DISCUSSION THREAD

**Discussion by Maxim-Romaniuk** [netbox-community/netbox#6845] [Q&A]
Hi! I want to create an application for the phone from which it will be possible to attach images to racks. How can I add images using REST API requests?
I thought the request would look like this:
```
{
  "content_type": "dcim.rack",
  "object_id": 19,
  "name": "someImgName",
  "image": "BASE64 code here",
  "image_height": 128,
  "image_width": 128
}
```
But it doesn't work:
```
{
  "image": [
    "The submitted data was not a file. Check the encoding type on the form."
  ]
}
```
Please tell me how to properly implement the attachment of an image from my application? 

P.S. I want to write an application in java, kotlin or using flutter.

**Comment by drygdryg** [comment]
Use `multipart/form-data` content type instead of JSON payload. It works for me:
```
curl "https://netbox.example.com/api/extras/image-attachments/" -H 'Authorization: Token abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMN' -F content_type=dcim.device -F object_id=236131 -F name="Test image" -F image_height=100 -F image_width=100 -F image=@test.jpg
```
